core/src/main/java/io/ddf/etl/TransformationHandler.java

```diff
@@ -361,12 +343,18 @@ public synchronized DDF transformUDFWithNames(String[] newColumnNames, String[]
 
     DDF newddf = this.getManager()
         .sql2ddf(sqlCmd, new SQLDataSourceDescriptor(sqlCmd, null, null, null, this.getDDF().getUUID().toString()));
+    newddf.getMetaDataHandler().copyFactor(this.getDDF());
```

Comment:
😅 move the implementation to `transformUDFWithNames` with `inPlace` flag and in this method call to that with `inPlace = false` is more natural IMO